spec/connection_spec.rb

```diff
@@ -92,4 +92,59 @@
 
   end
 
+  describe "jdbc_uri" do
+    before(:all) { @olap_connection = Mondrian::OLAP::Connection }
+
+    describe "MS SQL jTDS driver" do
+      it "should return a valid JDBC URI" do
+        @olap_connection.new(
+          driver: 'mssql',
+          host: 'example.com',
+          port: 1234,
+          instance: 'MSSQLSERVER',
+          database: 'example_db',
+          domain: 'win_domain'
+        ).jdbc_uri.should == 'jdbc:jtds:sqlserver://example.com:1234/example_db;instance=MSSQLSERVER;domain=win_domain'
+      end
+    end
+
+    describe "SQL Server" do
+      it "should return a valid JDBC URI with port" do
+        @olap_connection.new(
+          driver: 'sqlserver',
+          host: 'example.com',
+          port: 1234,
+          instance: 'MSSQLSERVER',
+          database: 'example_db'
+        ).jdbc_uri.should == 'jdbc:sqlserver://example.com:1234;databaseName=example_db'
+      end
+
+      it "should return a valid JDBC URI with instance name" do
+        @olap_connection.new(
+          driver: 'sqlserver',
+          host: 'example.com',
+          instance: 'MSSQLSERVER',
+          database: 'example_db'
+        ).jdbc_uri.should == 'jdbc:sqlserver://example.com\MSSQLSERVER;databaseName=example_db'
```

Comment:
I'm wondering that probably better to generate
```
jdbc:sqlserver://example.com;instanceName=MSSQLSERVER;databaseName=example_db
```
als in this case. Are there any reason why we should use `\MSSQLSERVER`?
It might be easier as well to implement it by transforming the `instance` parameter to the property `instanceName`.

Reply:
Checked how it is done in `activerecord-jdbc-adapter` and we shall do the same.